controllers/hyperparametertuningjob/hyperparametertuningjob_controller_test.go

```diff
@@ -18,1680 +18,538 @@ package hyperparametertuningjob
+			})
 
-		Expect(job.Status.TrainingJobStatusCounters.InProgress).ToNot(BeNil())
-		Expect(*job.Status.TrainingJobStatusCounters.InProgress).To(Equal(inProgress))
+			It("Updates status", func() {
+				ExpectStatusToBe(tuningJob, string(sagemaker.HyperParameterTuningJobStatusInProgress))
+			})
 
-		Expect(job.Status.TrainingJobStatusCounters.NonRetryableError).ToNot(BeNil())
-		Expect(*job.Status.TrainingJobStatusCounters.NonRetryableError).To(Equal(nonRetryableError))
+			Context("Spec defines HyperParameterTuningJobName", func() {
+				BeforeEach(func() {
+					tuningJob.Spec.HyperParameterTuningJobName = ToStringPtr("tuning-job-name")
+				})
 
-		Expect(job.Status.TrainingJobStatusCounters.RetryableError).ToNot(BeNil())
-		Expect(*job.Status.TrainingJobStatusCounters.RetryableError).To(Equal(retryableError))
+				It("Creates a HyperParameterTuningJob", func() {
+					req := receivedRequests.Front().Next().Value
+					Expect(req).To(BeAssignableToTypeOf((*sagemaker.CreateHyperParameterTuningJobInput)(nil)))
 
-		Expect(job.Status.TrainingJobStatusCounters.TotalError).ToNot(BeNil())
-		Expect(*job.Status.TrainingJobStatusCounters.TotalError).To(Equal(nonRetryableError + retryableError))
-		Expect(job.Status.TrainingJobStatusCounters.Stopped).ToNot(BeNil())
-		Expect(*job.Status.TrainingJobStatusCounters.Stopped).To(Equal(stopped))
+					createdRequest := req.(*sagemaker.CreateHyperParameterTuningJobInput)
+					Expect(*createdRequest.HyperParameterTuningJobName).To(Equal("tuning-job-name"))
+				})
+			})
+		})
 	})
 
-	It("should call the HpoTrainingJobSpawner", func() {
-
-		// Setup mock responses.
-		sageMakerClient := builder.
-			AddDescribeHyperParameterTuningJobResponse(description).
-			Build()
-
-		// Setup mock that reports the arguments it was called with
-		spawnMissingTrainingJobHpoJobs := List{}
-
-		mockHpoTrainingJobSpawner := callTrackingHpoTrainingJobSpawner{
-			SpawnMissingTrainingJobHpoJobs: &spawnMissingTrainingJobHpoJobs,
-		}
-
-		// Create controller with mocks and its request
-		controller := createHpoReconciler(k8sClient, CreateMockSageMakerClientProvider(sageMakerClient), createMockHpoTrainingJobSpawnerProvider(mockHpoTrainingJobSpawner), CreateMockAwsConfigLoader(), "1s")
-		request := CreateReconciliationRequest(job.ObjectMeta.Name, job.ObjectMeta.Namespace)
-
-		// Run test
-		controller.Reconcile(request)
-
-		// Verify that the hpoTrainingJobSpawner was called
-		Expect(spawnMissingTrainingJobHpoJobs.Len()).To(Equal(1))
+	Context("HyperParameterTuningJob exists", func() {
 
-		// Verify that the hpoTrainingJobSpawner received the correct parameters.
-		actualHpoJob := spawnMissingTrainingJobHpoJobs.Front().Value.(hpojobv1.HyperparameterTuningJob)
-		Expect(*actualHpoJob.Spec.HyperParameterTuningJobName).To(Equal(*job.Spec.HyperParameterTuningJobName))
-		Expect(*actualHpoJob.Spec.Region).To(Equal(*job.Spec.Region))
-		Expect(actualHpoJob.ObjectMeta.GetNamespace()).To(Equal(job.ObjectMeta.GetNamespace()))
```

Comment:
Why are we removing these assertions? We should verify that we invoke our mocked components correctly.